### PR TITLE
fix compile on mac os

### LINKED ISSUE #6651: Compile failed on mac os
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
compile on mac os

### 2. What did you expect to see? (Required)
Compile successfully.
### 3. What did you see instead (Required)
Compile failed.
https://cd.pingcap.net/blue/organizations/jenkins/build-common/detail/build-common/180835/pipeline/
https://cd.pingcap.net/blue/organizations/jenkins/build-common/detail/build-common/180823/pipeline/
### 4. What is your TiFlash version? (Required)
master a17c0bb2cd8c47e4ef48b881548dafd234f5ed42

### PR DESCRIPTION
**Pull request by hehechen**
Signed-off-by: hehechen <awd123456sss@gmail.com>

### What problem does this PR solve?

Issue Number: close #6651

Problem Summary:
fix compile on mac os
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Review by JaySon-Huang — APPROVED** [review]

**Review by SeaRise — APPROVED** [review]

File: dbms/src/Common/setThreadName.cpp [hunk 1]
```diff
@@ -77,5 +77,15 @@ std::string getThreadName()
 
 std::string getThreadNameAndID()
 {
-    return getThreadName() + "_" + DB::toString(pthread_self());
+    uint64_t thread_id;
+#if defined(__APPLE__)
```
Comment (CalvinNeo) [inline dbms/src/Common/setThreadName.cpp:81]: Can we just use `std::this_thread::get_id()`?

**Review by LittleFall — APPROVED** [review]

File: dbms/src/Common/setThreadName.cpp (hunk 1 above)
Comment (hehechen) [reply dbms/src/Common/setThreadName.cpp:81]: > Can we just use `std::this_thread::get_id()`?

`std::this_thread::get_id()` may not a uint64.  https://stackoverflow.com/questions/7432100/how-to-get-integer-thread-id-in-c11

**Pull request merged**